cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletEndpointTest.java

```diff
@@ -0,0 +1,269 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import kafka.utils.MockTime;
+import org.apache.kafka.common.utils.Time;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class KafkaCruiseControlServletEndpointTest {
+  private static final String GET_METHOD = "GET";
+  private static final String POST_METHOD = "POST";
+  private static final Function<String, OperationFuture> FUTURE_CREATOR = uuid -> new OperationFuture("future");
+
+  // A hack to make 2 requests to same endpoint 'look' different to UserTaskManager
+  private static final Map<String, String[]> DIFF_PARAM = new HashMap<>();
+  private static final Map<String, String[]> EMPTY_PARAM = Collections.emptyMap();
+
+  private static final UUID repeatUUID = UUID.randomUUID();
+
+  private static Collection<Object[]> _initializeServletRequestsOutput = new ArrayList<>();
+  private static Collection<Object[]> _populateUserTaskManagerOutput = new ArrayList<>();
+  private static Time _mockTime;
+  private static UserTaskManager.UUIDGenerator _mockUUIDGenerator;
+  private static HttpSession _mockHttpSession;
+  private static HttpServletResponse _mockHttpServletResponse;
+  private static UserTaskManager _userTaskManager;
+
+  private static final String[] PARAMS_TO_GET = {
+      ParameterUtils.CLIENT_IDS_PARAM,
+      ParameterUtils.ENDPOINTS_PARAM,
+      ParameterUtils.TYPES_PARAM,
+      ParameterUtils.USER_TASK_IDS_PARAM,
+      ParameterUtils.ENTRIES_PARAM
+  };
+
+  static {
+    DIFF_PARAM.put("param", new String[]{"true"});
+
+    _mockTime = new MockTime();
+    _mockUUIDGenerator = EasyMock.mock(UserTaskManager.UUIDGenerator.class);
+    _mockHttpSession = EasyMock.mock(HttpSession.class);
+    _mockHttpServletResponse = EasyMock.mock(HttpServletResponse.class);
+    EasyMock.expect(_mockHttpSession.getLastAccessedTime()).andReturn(_mockTime.milliseconds()).anyTimes();
+    _mockHttpSession.invalidate();
+    _mockHttpServletResponse.setHeader(EasyMock.anyString(), EasyMock.anyString());
+    EasyMock.expectLastCall().anyTimes();
+    _userTaskManager = new UserTaskManager(1000, 10, TimeUnit.HOURS.toMillis(6),
+        100, _mockTime, _mockUUIDGenerator);
+  }
+
+  private static class MockResult implements CruiseControlResponse {
+    public void discardIrrelevantResponse(CruiseControlParameters parameters) { }
+    public void writeSuccessResponse(CruiseControlParameters parameters, HttpServletResponse response) { }
+  }
+
+  private static Object[] inputRequestParams(UUID userTaskId, String clientId, String endPoint, Map<String, String[]> params,
+      boolean addToRequest, String methodType) {
```

Comment:
Bad indentation -- nit: for functions with a large number of arguments, we typically use one argument at each line.